File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) {
        return subarrays(nums,k) - subarrays(nums,k-1);
    }
    int subarrays(vector<int>& nums, int k){
        int l=0,r=0,n=nums.size(),count=0,sum=0;
        
        while( r< n){
            sum = sum + (nums[r]%2);
            while(sum > k){
                sum = sum - (nums[l] % 2);
                l++;
            }
            count = count + (r-l+1);
            r++;
        }
        return count;
    }
};

File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.java
class Solution {
    int func(int []nums,int goal){
        int l=0;
        int r=0;
        int sum=0;
        int cnt=0;
        int n=nums.length;
        if(goal<0){
            return 0;
        }
        while(r<n){
            sum+=(nums[r]%2);
            while(l<=r && sum>goal){
                sum-=(nums[l]%2);
                l++;
            }
            cnt+=(r-l+1);
            r++;
        }
        return cnt;
    }
    public int numberOfSubarrays(int[] nums, int k) {
        return func(nums,k)-func(nums,k-1);
    }
}